main/src/main/java/org/mobilitydata/gtfsvalidator/validator/NetworkIdConsistencyValidator.java

```diff
@@ -0,0 +1,175 @@
+    boolean hasNetworkIdField = this.routeTableContainer.hasColumn(GtfsRoute.NETWORK_ID_FIELD_NAME);
+
+    if (hasNetworkIdField && !this.routeNetworkTableContainer.isMissingFile()) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsRouteNetwork.FILENAME));
+    }
+    if (!this.networkTableContainer.isMissingFile() && hasNetworkIdField) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsNetwork.FILENAME));
+    }
+    validateUniqueRouteNetworkAssociation(noticeContainer);
+  }
+
+  private void validateUniqueRouteNetworkAssociation(NoticeContainer noticeContainer) {
+    Map<String, String> routeNetworkMap = buildRouteNetworkMap(noticeContainer);
+
+    for (GtfsRoute route : routeTableContainer.getEntities()) {
+      String routeId = route.routeId();
+      String networkId = route.networkId();
+      String existingRouteNetworkId = routeNetworkMap.get(routeId);
+
+      if (existingRouteNetworkId != null
+          && !existingRouteNetworkId.equals(networkId)
+          && networkId.length() > 0) {
+        int csvRowNumber =
+            routeNetworkTableContainer.byRouteIdMap().get(routeId).get(0).csvRowNumber();
+        noticeContainer.addValidationNotice(
+            new RouteNetworkAssociationDuplicateNotice(
+                routeId,
+                GtfsRouteNetwork.FILENAME,
+                csvRowNumber,
+                existingRouteNetworkId,
+                GtfsRoute.FILENAME,
+                route.csvRowNumber(),
+                networkId));
+      }
+    }
+  }
+
+  private Map<String, String> buildRouteNetworkMap(NoticeContainer noticeContainer) {
+    Map<String, String> routeNetworkMap = new HashMap<>();
+    for (Map.Entry<String, Collection<GtfsRouteNetwork>> entry :
+        routeNetworkTableContainer.byRouteIdMap().asMap().entrySet()) {
+      String routeId = entry.getKey();
+      Collection<GtfsRouteNetwork> routeNetworks = entry.getValue();
+      List<GtfsRouteNetwork> routeNetworksList = new ArrayList<>(routeNetworks);
+      if (routeNetworksList.size() > 1) {
+        for (int i = 1; i < routeNetworksList.size(); ++i) {
+          for (int j = i; j < routeNetworksList.size(); ++j) {
+            if (routeNetworksList.get(i).networkId().equals(routeNetworksList.get(j).networkId())) {
+              noticeContainer.addValidationNotice(
+                  new RouteNetworkAssociationDuplicateNotice(
+                      routeId,
+                      GtfsRouteNetwork.FILENAME,
+                      routeNetworksList.get(i).csvRowNumber(),
+                      routeNetworksList.get(i).networkId(),
+                      GtfsRouteNetwork.FILENAME,
+                      routeNetworksList.get(j).csvRowNumber(),
+                      routeNetworksList.get(j).networkId()));
```

Comment:
Instead of doing the full n x (n-1) cross product of duplicate entries, I think it might be more useful to just do the (n-1) duplicates.

Reply:
✅ Agreed and updated